openstack/containerinfra/v1/clustertemplates/requests.go

```diff
@@ -13,29 +13,33 @@ type CreateOptsBuilder interface {
 
 // CreateOpts params
 type CreateOpts struct {
-	Labels              map[string]string `json:"labels,omitempty"`
-	FixedSubnet         string            `json:"fixed_subnet,omitempty"`
+	Name                string            `json:"name,omitempty"`
+	COE                 string            `json:"coe" required:"true"`
+	ImageID             string            `json:"image_id" required:"true"`
+	FlavorID            string            `json:"flavor_id,omitempty"`
 	MasterFlavorID      string            `json:"master_flavor_id,omitempty"`
-	NoProxy             string            `json:"no_proxy,omitempty"`
-	HTTPSProxy          string            `json:"https_proxy,omitempty"`
-	HTTPProxy           string            `json:"http_proxy,omitempty"`
-	TLSDisabled         bool              `json:"tls_disabled,omitempty"`
+	DNSNameServer       string            `json:"dns_nameserver,omitempty"`
 	KeyPairID           string            `json:"keypair_id,omitempty"`
-	Public              bool              `json:"public,omitempty"`
-	DockerVolumeSize    int               `json:"docker_volume_size"`
-	ServerType          string            `json:"server_type,omitempty"`
 	ExternalNetworkID   string            `json:"external_network_id,omitempty"`
-	ImageID             string            `json:"image_id" required:"true"`
+	FixedNetwork        string            `json:"fixed_network,omitempty"`
+	FixedSubnet         string            `json:"fixed_subnet,omitempty"`
+	NetworkDriver       string            `json:"network_driver,omitempty"`
+	APIServerPort       int               `json:"apiserver_port,omitempty"`
+	DockerVolumeSize    int               `json:"docker_volume_size,omitempty"`
+	ClusterDistro       string            `json:"cluster_distro,omitempty"`
```

Comment:
Unfortunately, I think Magnum's API ref doc is not good and which is causing your misunderstanding here. Attributes like **ClusterDistro** is not required from the user side, it's extracted by Magnum from the given image.

Reply:
I would suggest just follow the openstackclient to figure out the parameters from user side instead of just using all the attributes at here https://github.com/openstack/magnum/blob/stable/queens/magnum/api/controllers/v1/cluster_template.py#L44 Cheers.

Reply:
For the fields, the interesting thing was, I was able to generate clustertemplates successfully both with and without the above four fields. Are you seeing something differently where these fields are required? Maybe we are running on different versions? I'll go ahead and remove the ClusterDistro. Yes the fields in the doc are incomplete and makes it confusing.

Reply:
BTW, feel free to make modifications